libs/librepcb/projecteditor/boardeditor/fsm/bes_addvia.cpp

```diff
@@ -198,9 +209,58 @@ bool BES_AddVia::entry(BEE_Base* event) noexcept {
   connect(mNetSignalComboBox, &QComboBox::currentTextChanged,
           [this](const QString& value) {
             setNetSignal(
-                mEditor.getProject().getCircuit().getNetSignalByName(value));
+                mEditor.getProject().getCircuit().getNetSignalByName(value)
+            );
           });
 
+  // add the "Start Layer:" label to the toolbar
+  mStartLayerLabel = new QLabel(tr("Start Layer:"));
+  mStartLayerLabel->setIndent(10);
+
+  // add the start layer combobox to the toolbar
+  mStartLayerComboBox = new QComboBox();
+  mStartLayerComboBox->setSizeAdjustPolicy(QComboBox::AdjustToContents);
+  mStartLayerComboBox->setInsertPolicy(QComboBox::NoInsert);
+  mStartLayerComboBox->setEditable(false);
+
+  // add the "Stop Layer:" label to the toolbar
+  mStopLayerLabel = new QLabel(tr("Stop Layer:"));
+  mStopLayerLabel->setIndent(10);
+
+  // add the start layer combobox to the toolbar
+  mStopLayerComboBox = new QComboBox();
+  mStopLayerComboBox->setSizeAdjustPolicy(QComboBox::AdjustToContents);
+  mStopLayerComboBox->setInsertPolicy(QComboBox::NoInsert);
+  mStopLayerComboBox->setEditable(false);
+
+  QString layerName = GraphicsLayer::sTopCopper;
+  mStartLayerComboBox->addItem(layerStack->getLayer(layerName)->getNameTr(),
+                               layerName);
+  for (int i = 1; i <= layerStack->getInnerLayerCount(); ++i){
+    QString layerName = GraphicsLayer::getInnerLayerName(i);
+    mStartLayerComboBox->addItem(layerStack->getLayer(layerName)->getNameTr(),
+                                 layerName);
+    mStopLayerComboBox->addItem(layerStack->getLayer(layerName)->getNameTr(),
+                                layerName);
+  }
+  layerName = GraphicsLayer::sBotCopper;
+  mStopLayerComboBox->addItem(layerStack->getLayer(layerName)->getNameTr(),
+                               layerName);
+  mStartLayerComboBox->setCurrentIndex(mCurrentVia->getStartLayerIndex());
+  mStopLayerComboBox->setCurrentIndex(mCurrentVia->getStopLayerIndex() - 1);
+  connect(mStartLayerComboBox,
+          static_cast<void (QComboBox::*)(int)>(&QComboBox::activated),
+          this, &BES_AddVia::startLayerChanged);
+  connect(mStopLayerComboBox,
+          static_cast<void (QComboBox::*)(int)>(&QComboBox::activated),
```

Comment:
> [signal] void QComboBox::activated(int index)
> This signal is sent when the user chooses an item in the combobox.

> [signal] void QComboBox::currentIndexChanged(int index)
> This signal is sent whenever the currentIndex in the combobox changes either through user interaction or **programmatically**.

I do change index programmically, so in case of currentIndexChanged(int), it will cause a loop.